### PR TITLE
Add cross compilation support for ARM and ARM64

### PR DESCRIPTION
**Pull request by benpye**
This adds cross compilation support for ARM and ARM64, including support in the build scripts, and a script to generate the rootfs for the selected architecture. Also included is some documentation for how to use the cross compilation support. This is currently two commits as I'm not sure if we want to remove all the comments in the tryrun files, or just the path sections?

@mmitche This will require root access once to generate the rootfs, you can set the location with the `ROOTFS_DIR` environment variable, you'd need to set it to somewhere where it will be preserved, and then set it when running `build.sh` for the cross compile too.

### CODE REVIEW

**Comment by benpye** [comment]
Failure isn't my fault, GitHub timeout.

**Comment by mmitche** [comment]
@benpye  I changed the job to shallow clone.  It could be github.  I'm seeing serious slowness with the website.

**Comment by mmitche** [comment]
@dotnet-bot test this please (retest github timeout)

File: cross/.gitmirror [hunk 1]
```diff
@@ -0,0 +1 @@
+Only contents of this folder, excluding subfolders, will be mirrored by the Git-TFS Mirror.
```
Comment (kangaroo) [inline cross/.gitmirror]: Do we really want to gitmirror the cross folder?

Comment (benpye) [inline cross/.gitmirror]: That should probably be a .gitmirrorall actually, as I also mirrored the other directories, I used a regular gitmirror as a rootfs directory may be generated there but it should never be committed, it's in .gitignore. Other Unix only components like the PAL are also under .gitmirror .

**Comment by mmitche** [comment]
Looks like github is still having a few issues.

@benpye Question...is there not enough cross compile support built into in CMake to do this without a rootfs? FYI I'm not particularly familiar with cross-compiling on *nix platforms.

**Comment by benpye** [comment]
@mmitche We still need the rootfs files for the include and lib directories to compile and link against, I agree it'd be nice to avoid, but I don't know a good alternative. What CMake provides is a clean(ish) way to override settings for different targets. Clang means we don't need a different compiler which is nice. I missed it from the docs (I'll add it now), the binutils for the target is also required. As far as I'm aware this is avoided on Windows as the Windows SDK includes the header files and libs for every supported target?

**Comment by mmitche** [comment]
@benpye Thanks for the info.

Yes, that's typically what happens.  The compilers are usually built already to target across multiple platforms, headers are typically shared and use the preprocessor to switch between platforms, and the libs are usually separated by directory.  The build system is then responsible for putting in the right defines and selecting the right LIB and INCLUDE directories.

**Comment by benpye** [comment]
Another thing I forgot to mention, this also fixes a build break on ARM64 that I accidentally caused in Oh another bb1d4ca

**Comment by mmitche** [comment]
@dotnet-bot test this please (another GH timeout)

**Comment by benpye** [comment]
Replaced a shell script with the `symlinks` command in the target rootfs, removes the need for a bit of the `build-rootfs.sh` script.

**Comment by benpye** [comment]
@dotnet-bot test this please

**Comment by mmitche** [comment]
@benpye, I'll take a look at this by the end of the day, but thanks for the effort.

**Comment by mmitche** [comment]
LGTM +1

**Pull request merged**